staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -25,40 +25,43 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager"
 )
 
 type attributesRecord struct {
-	kind        schema.GroupVersionKind
-	namespace   string
-	name        string
-	resource    schema.GroupVersionResource
-	subresource string
-	operation   Operation
-	options     runtime.Object
-	dryRun      bool
-	object      runtime.Object
-	oldObject   runtime.Object
-	userInfo    user.Info
+	kind         schema.GroupVersionKind
+	namespace    string
+	name         string
+	resource     schema.GroupVersionResource
+	subresource  string
+	operation    Operation
+	options      runtime.Object
+	dryRun       bool
+	object       runtime.Object
+	oldObject    runtime.Object
+	userInfo     user.Info
+	fieldManager *fieldmanager.FieldManager
 
 	// other elements are always accessed in single goroutine.
 	// But ValidatingAdmissionWebhook add annotations concurrently.
 	annotations     map[string]string
 	annotationsLock sync.RWMutex
 }
 
-func NewAttributesRecord(object runtime.Object, oldObject runtime.Object, kind schema.GroupVersionKind, namespace, name string, resource schema.GroupVersionResource, subresource string, operation Operation, operationOptions runtime.Object, dryRun bool, userInfo user.Info) Attributes {
+func NewAttributesRecord(object runtime.Object, oldObject runtime.Object, kind schema.GroupVersionKind, namespace, name string, resource schema.GroupVersionResource, subresource string, operation Operation, operationOptions runtime.Object, dryRun bool, userInfo user.Info, fieldManager *fieldmanager.FieldManager) Attributes {
```

Comment:
It feels wrong to add fieldManager here, because all the other pieces of attributesRecord are concerning a specific request, but the field Manager is defined per resource. Not sure where would be a better place for this though.

Reply:
@jennybuckley locally I reverted the commit and extended the OpbjectInterfaces with a GetObjectFieldManager function as it is scope anyways in almost all cases. This is a lot smaller. Maybe that's better.

Thanks for the suggestions above, I'll look into them right today.